Folder structure changes and renaming

zhenwendai opened the pull request:
It contains the following changes:
- Folder structure changes and renaming according to our discussion.
- Extend the Mark's multivariate normal distribution implementation to allow general broadcasting.

@marpulli I changed the behavior of your log_pdf and kl_divergence implementation to return an array of the size of mean without the last dimension. This is consistent with our other distributions.


By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

meissnereric commented on mxfusion/runtime/distributions/distribution.py:
```diff
@@ -13,7 +13,7 @@
 
 
-class DistributionImplementation(object):
+class RuntimDistribution(object):
```
RunTimDistribution is misspelled (RunTim -> RunTime)

meissnereric commented on mxfusion/runtime/distributions/multivariate_normal.py:
```diff
@@ -12,38 +12,50 @@
 
-        if mean.ndim != 3:
-            raise ValueError('Mean should have 3 dimensions. It has {}'.format(mean.ndim))
+        # if mean.ndim != 3:
```
Either remove these commented out lines or add them back in please.

marpulli commented:
At the minute there is
`mxfusion.components.distributions.bernoulli.Bernoulli` and 
`mxfusion.runtime.distributions.bernoulli.Bernoulli`

It might be confusing to have two different classes with the same name in the same library.

Can we do something like `mxfusion.runtime.distributions.bernoulli.BernoulliRuntime` or 
`mxfusion.components.distributions.BernoulliFactor` to distinguish them?

zhenwendai commented:
- I have renamed all the run time distributions as XXXRunTime to distinguish from the model definition distributions and leave the name of original distributions unchanged to keep interface consistency.

- I implemented the mechanism that, if the run time distribution of a model definition distribution is specified, it uses the run time distribution for the log_pdf and draw_samples computation.
